docs/preview/02-Features/02-message-handling/01-service-bus.md

```diff
@@ -140,111 +131,105 @@ public class Startup
-        services.AddServiceBusQueueMessagePump(
-            "My-Service-Bus-Queue-Name",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
+        services.WithServiceBusMessageHandler<OrderMessageHandler, Order>(context => context.Properties["MessageType"].ToString() == "Order");
+    }
+}
+```
 
-        // Specify the name of the Service Bus Topic, and provide a name for the Topic subscription:
-        services.AddServiceBusMessageTopicMessagePump<OrdersMessageHandler>(
-            "My-Service-Bus-Topic-Name",
-            "My-Service-Bus-Topic-Subscription-Name",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
+> Note that the order in which the message handlers are registered is important in the message processing.
+> In the example, when a message handler above this one is registered that could also handle the message (same message type) than that handler may be chosen instead of the one with the specific filter.
 
-        // Specify a topic subscription prefix instead of a name to separate topic message pumps.
-        services.AddServiceBusTopicMessagePumpWithPrefix(
-            "My-Service-Bus-Topic-Name"
-            "My-Service-Bus-Subscription-Prefix",
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING");
 
-        // Uses managed identity to authenticate with the Service Bus Topic:
-        services.AddServiceBusTopicMessagePumpUsingManagedIdentity(
-            topicName: properties.EntityPath,
-            subscriptionName: "Receive-All", 
-            fullyQualifiedNamespace: "<your-namespace>.servicebus.windows.net"
-            // The optional client id to authenticate for a user assigned managed identity. More information on user assigned managed identities cam be found here:
-            // https://docs.microsoft.com/en-us/azure/active-directory/managed-identities-azure-resources/overview#how-a-user-assigned-managed-identity-works-with-an-azure-vm
-            clientId: "<your-client-id>");
+### Bring your own deserialization
+You can also choose to extend the built-in message deserialization with additional deserializer to meet your needs. 
+This allows you to easily deserialize into different message formats or reuse existing (de)serialization capabilities that you already have without altering the message router. 
 
-        services.AddServiceBusTopicMessagePump(
-            "ARCUS_SERVICEBUS_ORDERS_CONNECTIONSTRING",
-            options => 
-            {
-                // Indicate whether or not messages should be automatically marked as completed 
-                // if no exceptions occurred and processing has finished (default: true).
-                options.AutoComplete = true;
+You start by implementing an `IMessageBodySerializer`. The following example shows how an expected type can be transformed to something else. 
+The result type (in this case `OrderBatch`) will be then be used to check if there is an `IAzureServiceBusMessageHandler` registered with that message type.
```

Comment:
```suggestion
The result type (in this case `OrderBatch`) will then be used to check if there is an `IAzureServiceBusMessageHandler` registered for that message type.
```

Reply:
Done in https://github.com/arcus-azure/arcus.messaging/pull/268/commits/cd44eb2753ac33d70bef057cea45e54bbf30fb40